google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/cloudpubsub/internal/WrappingPublisher.java

```diff
@@ -55,6 +57,15 @@ protected void handlePermanentError(StatusException error) {}
   // Publisher implementation.
   @Override
   public ApiFuture<String> publish(PubsubMessage message) {
+    ApiService.State currentState = state();
```

Comment:
I don't think this is the right layer to enforce this at. The WrappingPublisher is just a thin wrapper around the internal publisher. Perhaps add this in the wire publisher implementation instead?

Reply:
Done. Moved the check to PublisherImpl.